File: Two-Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        \\\
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        \\\
        num_map = {} 

        for i, num in enumerate(nums):
            c = target - num
            if c in num_map:
                return [num_map[c], i]
            num_map[num] = i

        return []  

File: Two-Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map(); 

    for (let i = 0; i < nums.length; i++) {
        const c = target - nums[i];
        if (map.has(c)) {
            return [map.get(c), i];
        }
        map.set(nums[i], i);
    }

    return []; 
};
